### PR TITLE
Added footer-prop to Modal and a ModalFooter component (UIEH-434)

### PR DESCRIPTION
**Pull request by rasmuswoelk**
## Modal changes
This update adds a footer-prop for the Modal-component that will accept anything to be passed into it — allowing developers to make their own modal footer content if needed.

The default styling of the modal footer within the `<Modal>` consists of the default gutter and some flexbox properties. Content is set to flex-end using flexbox' justify-content. This works with RTL-direction. 

I have noted in the readme that it's recommended to use React's `<Fragment>` component to pass content in order to preserve the alignment of the footer content without having to write additional CSS. This is of course only a recommendation and not a requirement.

I have updated both readme and examples for the `<Modal>`-component and they are available in Storybook.

## ModalFooter
image

In addition to the footer-prop on the `<Modal>` I have added a ModalFooter-component which will be used as a default modal footer. This should make the rendered UI of the modals more consistent and even save time for developers while making basic modals for their apps.

It renders a primary and a secondary `<Button>` with the appropriate props. This matches most of our use-cases currently and it can easily be extended to support more content if needed.

I have added a ModalFooter example which can be found under the `<Modal>`-section in Storybook.

## To-do

- [ ] Add tests for `<ModalFooter>` (in a new PR once this one is approved)
- [ ] Use  `<ModalFooter>` in `<ConfirmationModal>`
- [ ] Use ConfirmationModal in `<StripesFormModal>` in `stripes-form`
- [ ] Fix https://issues.folio.org/browse/UIEH-342

## Relevant issues
https://issues.folio.org/browse/UIEH-434

### CODE REVIEW

**Comment by doytch** [comment]
Sliiiick! Any thoughts about enabling developer laziness even further to make it so that if I want to add a ModalFooter with Save/Cancel buttons, all I need to do is define like, `onSave` and `onCancel` props on a Modal? Maybe along with an `includeFooter` boolean? 😆 

It might clash with other props so I'm not sure...

**Comment by doytch** [comment]
Though then again, we're not consistent with Save/Cancel. Sometimes we have other labels that we use when it's more appropriate so this is probably better. :D

**Comment by rasmuswoelk** [comment]
@doytch For that, you would use the ConfirmationModal-component which implements those exact props: https://github.com/folio-org/stripes-components/tree/master/lib/ConfirmationModal 👍 

In general, I think we are trying to spread out the functionality a bit so that components won't become bloated with a lot of stuff that isn't needed for the broad use-cases. 

E.g. for this case we provide an easy way of displaying a default-footer by having a component for it while still preserving the option to make an entirely new ModalFooter that fits a specific use-case. Hopefully developers (being lazy and all..) will choose to use our modal footer if it fits their needs thus keeping the design consistent. 👍

**Comment by cherewaty** [comment]
👍 this looks excellent

@wwilsman any interactor suggestions here?

File: lib/ModalFooter/tests/ModalFooter-test.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      />
+    );
+  });
+
+  /**
+   * The "primaryButton"-prop should render a primary <Button>  with the primary button style
+   */
+  it('renders a primary button when using the primaryButton-prop', () => {
+    const button = modalFooter.$(`#${primaryButtonId}`);
+    expect(button.className).to.include(buttonCss.primary);
```
[the remark this answers is not in the document]
Comment (rasmuswoelk) [reply lib/ModalFooter/tests/ModalFooter-test.js]: Got it.

File: lib/ModalFooter/tests/ModalFooter-test.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  });
+
+  /**
+   * The "secondaryButton"-prop should render a <Button> with the default button style
+   */
+  it('renders a default button when using the secondaryButton-prop', () => {
+    const button = modalFooter.$(`#${secondaryButtonId}`);
+    expect(button.className).to.include(buttonCss.default);
```
[the remark this answers is not in the document]
Comment (rasmuswoelk) [reply lib/ModalFooter/tests/ModalFooter-test.js]: 👍

File: lib/ModalFooter/tests/interactor.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/**
+ * ModalFooter interactor
+ */
+
+import { interactor } from '@bigtest/interactor';
+
+export default interactor(class ModalFooterInteractor {
```
[the remark this answers is not in the document]
Comment (rasmuswoelk) [reply lib/ModalFooter/tests/interactor.js]: You are totally right. I will remove it.

**Comment by rasmuswoelk** [comment]
@wwelling Sorry that I merged this one in before you had a chance to review but it had been stuck in review for a few days and I really needed these updates to continue my work.

Thanks for your feedback on the tests. I will fix them ASAP 👍

File: lib/ModalFooter/tests/interactor.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (wwilsman) [reply lib/ModalFooter/tests/interactor.js]: Looking back over my comments, I realized all of these tests are related to the `button`. We could nest the button interactor inside this modal footer and reuse them.

```js
import { interactor, scoped } from '@bigtest/interactor';
import ButtonInteractor from '../../Button/tests/interactor';

export default interactor(class ModalFooterInteractor {
  primaryButton = scoped(`#${primaryButtonId}`, ButtonInteractor);
  secondaryButton = scoped(`#${secondaryButtonId}`, ButtonInteractor);
}
```

And used like

``` js
expect(modalFooter.primaryButton.rendersPrimary).to.be.true
expect(modalFooter.secondaryButon.label).to.equal(secondaryButtonLabel)
```

[the remark this answers is not in the document]
Comment (rasmuswoelk) [reply lib/ModalFooter/tests/interactor.js]: Cool! Still getting into @bigtest so I'm grateful for your feedback. The above looks like a better way to go!

[the remark this answers is not in the document]
Comment (wwilsman) [reply lib/ModalFooter/tests/interactor.js]: There are some proper guides for this coming soon :)

**Pull request merged**